ocaml/xcp-rrdd/bin/rrdd/rrdd_monitor.ml

```diff
@@ -138,31 +138,30 @@ let update_rrds timestamp dss uuid_domids paused_vms =
         in
         let dss = StringMap.to_seq dss |> Seq.map snd |> List.of_seq in
 
-        try
-          let domid = StringMap.find vm_uuid uuid_domids in
+        match StringMap.find_opt vm_uuid uuid_domids with
+        | Some domid -> (
           (* First, potentially update the rrd with any new default dss *)
-          try
-            let rrdi = Hashtbl.find vm_rrds vm_uuid in
-            let rrd = merge_new_dss rrdi.rrd dss in
-            Hashtbl.replace vm_rrds vm_uuid {rrd; dss; domid} ;
-            (* CA-34383: Memory updates from paused domains serve no useful
-               purpose. During a migrate such updates can also cause undesirable
-               discontinuities in the observed value of memory_actual. Hence, we
-               ignore changes from paused domains: *)
-            if not (StringSet.mem vm_uuid paused_vms) then (
-              Rrd.ds_update_named rrd timestamp ~new_domid:(domid <> rrdi.domid)
-                named_updates ;
-              rrdi.dss <- dss ;
-              rrdi.domid <- domid
-            )
-          with
-          | Not_found ->
+          match Hashtbl.find_opt vm_rrds vm_uuid with
+          | Some rrdi ->
+              let rrd = merge_new_dss rrdi.rrd dss in
+              Hashtbl.replace vm_rrds vm_uuid {rrd; dss; domid} ;
+              (* CA-34383: Memory updates from paused domains serve no useful
+                 purpose. During a migrate such updates can also cause undesirable
+                 discontinuities in the observed value of memory_actual. Hence, we
+                 ignore changes from paused domains: *)
+              if not (StringSet.mem vm_uuid paused_vms) then (
+                Rrd.ds_update_named rrd timestamp
+                  ~new_domid:(domid <> rrdi.domid) named_updates ;
+                rrdi.dss <- dss ;
+                rrdi.domid <- domid
+              )
+          | None ->
               debug "%s: Creating fresh RRD for VM uuid=%s" __FUNCTION__ vm_uuid ;
               let rrd = create_fresh_rrd !use_min_max dss in
               Hashtbl.replace vm_rrds vm_uuid {rrd; dss; domid}
-          | e ->
-              raise e
-        with _ -> log_backtrace ()
+        )
+        | None ->
+            log_backtrace ()
```

Comment:
There's no backtrace to collect here, this should have a new log message

Reply:
What would be the best log message here? Struggling to think of something that describes the logic well

Reply:
Hmm, this could work:
```suggestion
info "%s: VM %s is not resident in this host, ignoring" __FUNCTION__
```
If it starts logspamming we have a bug to fix, but this would have been worse with the backtraces that the existing code produces.